util/Migrator/DbScripts/2022-12-26_00_OrganizationUserReadByUserIdWithPolicyDetails.sql

```diff
@@ -0,0 +1,46 @@
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+
+CREATE OR ALTER PROCEDURE [dbo].[OrganizationUser_ReadByUserIdWithPolicyDetails]
+    @UserId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+SELECT
+    P.[OrganizationId],
+    P.[Type] as PolicyType,
+    P.[Enabled] as PolicyEnabled,
+    P.[Data] as PolicyData,
+    OU.[Type] as OrganizationUserType,
+    OU.[Status] as OrganizationUserStatus,
+    (CASE WHEN OU.[Permissions] IS NULL THEN 0 ELSE (CASE WHEN COALESCE(JSON_VALUE(OU.[Permissions], '$.managePolicies'), 'false') = 'false' THEN 0 ELSE 1 END) END) as CanManagePolicies,
+    (CASE WHEN PUPO.[UserId] IS NULL THEN 0 ELSE 1 END) as IsProvider
+FROM
+    [dbo].[PolicyView] P
+INNER JOIN
+    [dbo].[OrganizationUserView] OU ON P.[OrganizationId] = OU.[OrganizationId]
+LEFT JOIN
+    (SELECT
+        PU.UserId,
+        PO.OrganizationId
+    FROM
+        [dbo].[ProviderUserView] PU
+    INNER JOIN
+        [ProviderOrganizationView] PO ON PO.[ProviderId] = PU.[ProviderId]) PUPO
+    ON PUPO.UserId = OU.UserId
+    AND PUPO.OrganizationId = P.OrganizationId
```

Comment:
I don't like this nested query/join but I don't know if we can avoid it. Any ideas?

Reply:
I rewrote the query, let me know what you think.